### PR TITLE
[GO] add completion spec

### PR DESCRIPTION
**Pull request by cstrnt**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the current behavior? (You can also link to an open issue here)**

There is no completion spec for `go`

**What is the new behavior (if this is a feature change)?**
There will be one 🥳 

**Additional info:**

### CODE REVIEW

File: dev/go.ts [hunk 1]
```diff
@@ -0,0 +1,783 @@
...
+          name: "-t",
+          description:
+            "modules needed to build tests of packages specified on the command line.",
+        },
+        {
+          name: "-u",
+          description: "update to newer minor or patch releases when available",
+        },
+        {
+          name: "-u=patch",
```
Comment (cstrnt) [inline dev/go.ts:387]: FIX ME

File: specs/go.js [hunk 2]
```diff
@@ -0,0 +1,720 @@
...
+                },
+                {
+                    name: "-deps",
+                    description: "terate over not just the named packages but also all their dependencies",
+                },
+                {
+                    name: "-f",
+                    insertValue: "-f '{cursor}'",
+                    description: "specify an alternate format for te list",
+                    args: {
```
Comment (mattschrage) [inline specs/go.js:393]: Typo.

**Review by mattschrage — COMMENTED** [review]
Everything seems fine to me. Once @falky97 takes a look, I think it's ready to merge in.

File: dev/go.ts (hunk 1 above)
Comment (cstrnt) [reply dev/go.ts:387]: Actually not, it's a feature!

File: dev/go.ts [hunk 3]
```diff
@@ -0,0 +1,783 @@
...
+          name: "-n",
+          description: "print the commands that would be executed",
+        },
+        {
+          name: "-x",
+          description: "print the commands as they are exectued",
+        },
+        {
+          name: "-mod",
+          description: "which module download modeto use",
```
Comment (cstrnt) [inline dev/go.ts:343]: typo

File: dev/go.ts (hunk 1 above)
Comment (falky97) [reply dev/go.ts:387]: Fig's tokenizer will treat this as [-u, patch]. So we should add an optional argument to `-u`

File: specs/go.js [hunk 4]
```diff
@@ -0,0 +1,720 @@
...
+        },
+    },
+    {
+        name: "-trimpath",
+        description: "remove all file system paths from the resulting executable.",
+    },
+]);
+var packagesArg = {
+    name: "packages",
+    variadic: true,
```
Comment (falky97) [inline specs/go.js:194]: `isOptional: true`

File: dev/go.ts [hunk 5]
```diff
@@ -0,0 +1,783 @@
...
+      options: [
+        ...globalOptions,
+        {
+          name: "-exec",
+          description: "invoke the binary using xprog",
+          args: {},
+        },
+      ],
+      args: {
+        name: "package",
```
Comment (falky97) [inline dev/go.ts:683]: Look into this more, but likely would add the prop: `isScript: true`

**Review by QuiiBz — APPROVED** [review]

**Comment by falky97** [comment]
Going to merge this in. I know we went over this together. Given how poor go's documentation is I anticipate there are going to be some problems here. Here's hoping we have some go aficionados come in and give us a helping hand :)

**Pull request merged**